src/types/onyx/TransactionViolation.ts

```diff
@@ -92,6 +92,9 @@ type TransactionViolation = {
 
     /** Additional violation information to provide the user */
     data?: TransactionViolationData;
+
+    /** Indicates if this violation should be shown in review */
+    showInReview?: boolean;
```

Comment:
Does this need to be optional?

Reply:
Yes, because I am not sending it from PHP when it is not `true`

Reply:
```suggestion
    showInReview: boolean | undefined;
```
This should be required to be set (and still allow for undefined). This is needed so we create the correct optimistic data. Not doing so led to a regression https://github.com/Expensify/App/issues/54510

However, now we need to know which violations has `showInReview=true`. Can you please provide a list?

Reply:
Why would this need to be set for optimistic actions to work? 😕 

> However, now we need to know which violations has showInReview=true. Can you please provide a list?

I don't think there's a list, there's a bunch of logic controlling this property's value.

Reply:
> Why would this need to be set for optimistic actions to work?

If we don't then expenses created offline that violate policy rules won't have the red dot set on their previews. This is due to this logic https://github.com/Expensify/App/blob/ad99c20c40e4fed347cff29f5ed85c98825442a8/src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx#L120 that only take into account violations with `showInReview=true`

Reply:
Ah ok, so let's add that property to the optimistic actions (assuming the property will be removed when the data from the server is returned), but no need to mark this prop as required

Reply:
The problem with not making this required is that the bug could resurface (one could simply forget to add the field). Do you think we shouldn't optimize for this case?

Reply:
Do you think it actually prevents bugs? Allowing undefined gives the same signal to code authors as allowing it to not be passed, no?

Reply:
I see your point. Forcing the field to be set can reduce the occurrence of such bugs as you will get a chance to put in the correct value but it won't 100% prevent the bug.

We will keep the types untouched.